src/socket/socket_client.js

```diff
@@ -0,0 +1,107 @@
+
+const EventEmitter = require('events').EventEmitter
+const util = require('./util')
+
+const TCPSocketClient = require('./socket_client_tcp')
+const WebSocketClient = require('./socket_client_ws')
+
+class SocketClient {
+  constructor(host, port, protocol, options) {
+    this.id = 0
+    this.host = host
+    this.port = port
+    this.protocol = protocol
+    this.options = options
+    this.callback_message_queue = {}
+    this.subscribe = new EventEmitter()
```

Comment:
I think that this makes more sense to be `.events`, based on its type. What's the rationale for the rename?

Reply:
Yeah, I like `events` better too. But we had to fork a base library where it's `subscribe`. I didn't want to diverge too much from the base repo, but later refactored it significantly anyways so I think this rename from `subscribe` to `events` won't be a huge change compared to all others. 
I'd like to propose we do this rename later, as we have the current solution tested and confirmed on both your and my machines. I plan to do more refactoring after the demo so will pick up this rename too.

Reply:
Created an issue for this change: https://github.com/nkuba/electrum-client-js/issues/3